eclipse-scout-core/src/desktop/outline/pages/js/JsPageHelper.ts

```diff
@@ -0,0 +1,470 @@
+
+  objectType: string;
+
+  page: Page;
+
+  protected _replaceChildPages = true;
+  /**
+   * Child pages by id (see {@link #_createChildPageId}).
+   */
+  protected _childPagesById = new Map<string, JsPageChildPage>();
+
+  // general listeners
+
+  protected _nodesUpdatedHandler = this._onNodesUpdated.bind(this);
+  protected _nodesDeletedHandler = this._onNodesDeleted.bind(this);
+  protected _allChildNodesDeletedHandler = this._onAllChildNodesDeleted.bind(this);
+
+  // listeners for PageWithTable
+
+  protected _nodesInsertedHandler = this._onNodesInserted.bind(this);
+  protected _tableRowsInsertHandler = this._onTableRowsInserted.bind(this);
+
+  init(model: InitModelOf<this>) {
+    $.extend(this, model);
+
+    scout.assertInstance(this.page, Page, 'Page not set or has wrong type');
+
+    // do not replace child pages initially
+    // if e.g. the browser is reloaded the previously created child pages already exist on the UI server and therefore childNodes are set in the pages initModel
+    // if the page is a PageWithTable its table will be loaded when the page is selected and this load must not lead to all child pages being replaced
+    this._replaceChildPages = !this.page.childNodes?.length;
+
+    this._installListeners();
+    this._installPageWithTableListeners();
+  }
+
+  public destroy() {
+    this._uninstallListeners();
+    this._uninstallPageWithTableListeners();
+    this.page = null;
+  }
+
+  protected _installListeners() {
+    this.page.outline.on('nodesUpdated', this._nodesUpdatedHandler);
+    this.page.outline.on('nodesDeleted', this._nodesDeletedHandler);
+    this.page.outline.on('allChildNodesDeleted', this._allChildNodesDeletedHandler);
+  }
+
+  protected _uninstallListeners() {
+    this.page.outline.off('nodesUpdated', this._nodesUpdatedHandler);
+    this.page.outline.off('nodesDeleted', this._nodesDeletedHandler);
+    this.page.outline.off('allChildNodesDeleted', this._allChildNodesDeletedHandler);
+  }
+
+  protected _onNodesUpdated(event: TreeNodesUpdatedEvent) {
+    if (!event.nodes?.length) {
+      return;
+    }
+
+    // filter child pages of page
+    const childPages = event.nodes.filter(node => node.parentNode === this.page) as Page[];
+    if (!childPages.length) {
+      return;
+    }
+
+    // Child pages may have been updated with e.g. a new text. Send a nodesChanged event in order to send this information to the UI server.
+    // This information is needed when the browser is reloaded as the child pages will still be available.
+    (this.page.outline.modelAdapter as OutlineAdapter).sendNodesChanged(childPages);
+  }
+
+  protected _onNodesDeleted(event: TreeNodesDeletedEvent) {
+    if (!event.nodes?.length) {
+      return;
+    }
+
+    // mark page as childrenLoaded=false if all child pages are deleted
+    if (!this.page.childNodes.length) {
+      this.page.childrenLoaded = false;
+    }
+
+    if (!this._childPagesById.size) {
+      return;
```

Comment:
Is it guaranteed that the detailTable does not have any rows if this map is empty?